### ISSUE TITLE
Generic.WhiteSpaceScopeIndent closure argument indenting incorrect with multi-line strings

### ISSUE THREAD

**Issue by turboatcrazyfactory** [squizlabs/PHP_CodeSniffer#1637]
Hello,
I'm using Sniffer to lint the test code which is using Codeception\Specify in the code.
# Problem #
When doing this:
```php
class Dummy extends Unit
{
    use Specify;

    public function foo()
    {
        $this->specify('some
        long description', function () {
            echo 'hello';
        });
    }
}
```
After running phpcbf
```php
class Dummy extends Unit
{
    use Specify;

    public function foo()
    {
        $this->specify('some
        long description', function () {
            echo 'hello';
}); // <----- wrong indent
    }
}
```

# What I expected #
Sniff should not adjust the indent of the first code.

# Solution ? #
Should we include `T_CONSTANT_ENCAPSED_STRING` when finding the first token position of the line ?

```php
// ScopeIndentSniff.php, Line 589

if (isset($tokens[$scopeCloser]['scope_condition']) === true) {
        $first = $phpcsFile->findFirstOnLine([T_WHITESPACE, T_CONSTANT_ENCAPSED_STRING], $tokens[$scopeCloser]['scope_condition'], true);

        $currentIndent = ($tokens[$first]['column'] - 1);
// ...

**Comment by gsherwood** [comment]
Shorter code to replicate:
```php
<?php
public function foo()
{
    $foo('some
    long description', function () {
    });
}
```
Scope indent debug output:
```
Start with token 0 on line 1 with indent 0
Closing parenthesis found on line 2
	 * ignoring single-line definition *
Open scope (T_FUNCTION) on line 3
	=> indent set to 4 by token 9 (T_OPEN_CURLY_BRACKET)
Open closure on line 5
	=> indent set to 4 by token 23 (T_OPEN_CURLY_BRACKET)
Close scope (T_CLOSURE) on line 6
	=> indent set to 0 by token 26 (T_CLOSE_CURLY_BRACKET)
[Line 6] Line indented incorrectly; expected 0 spaces, found 4
	=> Add adjustment of -4 for token 26 (T_CLOSE_CURLY_BRACKET) on line 6
Close closure on line 6
	* token has nested parenthesis 13 on line 4 *
	* token is inside condition 3 (T_FUNCTION) on line 2 *
	* using parenthesis *
	* previous token is T_VARIABLE on line 4 *
	* first token on line 4 is 12 (T_VARIABLE) *
	=> indent set to 4 by token 12 (T_VARIABLE)
Closing parenthesis found on line 6
	* token is inside condition 3 (T_FUNCTION) on line 2 *
	* using condition *
	=> checking indent of 4; main indent set to 4 by token 3 (T_FUNCTION)
Close scope (T_FUNCTION) on line 7
	=> indent set to 0 by token 30 (T_CLOSE_CURLY_BRACKET)
```
Problem is here:
```
Open closure on line 5
	=> indent set to 4 by token 23 (T_OPEN_CURLY_BRACKET)
```
The indent should be set to 8, but the first non-whitespace token on the line is a multi-string at column 0, so it can't be used to determine indent in this case. The indent checks needs to go back another line to find the true indent.

**Comment by gsherwood** [comment]
I've committed a fix for this issue, which required checks for multi-line strings when opening and closing closures. Thanks for reporting it.
